src/server/database/Database/Transaction.cpp

```diff
@@ -67,16 +69,26 @@ bool TransactionTask::Execute()
 {
     int errorCode = m_conn->ExecuteTransaction(m_trans);
     if (!errorCode)
+    {
+        m_trans->Cleanup();
         return true;
+    }
 
     if (errorCode == ER_LOCK_DEADLOCK)
     {
         // Make sure only 1 async thread retries a transaction so they don't keep dead-locking each other
-        std::lock_guard<std::mutex> lock(_deadlockLock);
+        //std::lock_guard<std::mutex> lock(_deadlockLock);
         uint8 loopBreaker = 5;  // Handle MySQL Errno 1213 without extending deadlock to the core itself
         for (uint8 i = 0; i < loopBreaker; ++i)
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(24 + rand() % 24));
```

Comment:
how do you expect randomness to fix a issue ? just by pure luck ?

Reply:
reading more in detail https://github.com/TrinityCore/TrinityCore/issues/20106 I would agree that giving more time to other SQL transactions to finish should help resolve the deadlock, I would prefer to change the loop from looping only 5 times to loop for a configurable amount of time, default to 30/60 seconds

Reply:
The randomness was added just in case 2 threads are conflicting at the same time that they are offset from each other, and given time to complete.

Reply:
can we add a sleep with the lock enabled, just so it doesn't hammer SQL requests too often.